pkg/operator/controller/controller_lb.go

```diff
@@ -107,6 +105,19 @@ func (r *reconciler) currentLoadBalancerService(ci *ingressv1alpha1.ClusterIngre
 	return service, nil
 }
 
+// enforceLoadBalancerServiceFinalizer adds loadBalancerServiceFinalizer to service
+// if it doesn't exist.
+func (r *reconciler) enforceLoadBalancerServiceFinalizer(service *corev1.Service) error {
```

Comment:
I'm not clear why this was extracted out of the 1-liner assignment as part of the create. The reason we need enforcement for the clusteringress is because we can't mutate the incoming resource during admission, so we must update it after the fact. The operator owns the creation of the service, and so can set the finalizer atomically on the service during create. Therefore there's no need to ever check/update again.

Reply:
OK, I can revert. I thought we wanted to "enforce" the finalizer. Previously, the finalizer could be removed by a user.

Reply:
I think we can enforce that in validation once the API refactor is done